### ISSUE TITLE
Experiment 2.5

### ISSUE THREAD

**Issue by jds8** [jds8/SAE4DLM-CE#37]
In this experiment, we aim to test how (multiplicative) randomness affects CE loss. In particular, at each layer (L1, L5, L10, L14, L23, L27), we should add a forward hook, like the SAE hook, that multiplies the layer activations by a random matrix. We still want to generate the usual JSON files containing CE delta results.

We should scale the column norms of the random matrix to equal those of a trained SAE. We can do this using:

```
trained_sae = AutoEncoderTopK.from_pretrained(...) # as in experiment 2
with torch.no_grad():
    trained_matrix = trained_sae.encoder.weight @ trained_sae.decoder.weight
random_matrix = torch.randn_like(trained_matrix)
scaled_matrix = random_matrix * trained_matrix.norm(dim=0, keepdim=True)
```

Note that to load the trained SAE, you'll have to choose an arbitrary trainer/sparsity level. It doesn't matter which.
